libbpf-rs/tests/common/mod.rs

```diff
@@ -116,9 +126,18 @@ pub fn get_symbol_offset(binary_path: &Path, symbol_name: &str) -> Result<usize,
         }
         if let Some(name) = elf.strtab.get_at(sym.st_name) {
             if name == symbol_name {
-                let sec = &elf.section_headers[sym.st_shndx];
-                let offset = sym.st_value - sec.sh_addr + sec.sh_offset;
-                return Ok(offset as usize);
+                for phdr in elf.program_headers.iter() {
```

Comment:
Can we find a way to eliminate duplication with the above? It seems we should be able to factor out a function?